hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -155,46 +155,68 @@ int SaraNcpClient::initParser(Stream* stream) {
             .commandTerminator(AtCommandTerminator::CRLF);
     parser_.destroy();
     CHECK(parser_.init(std::move(parserConf)));
+    // +CREG: <stat>[,<lac>,<ci>[,<AcTStatus>]]
     CHECK(parser_.addUrcHandler("+CREG", [](AtResponseReader* reader, const char* prefix, void* data) -> int {
         const auto self = (SaraNcpClient*)data;
-        int val[2];
-        int r = CHECK_PARSER_URC(reader->scanf("+CREG: %d,%d", &val[0], &val[1]));
+        int val[4];
+        // Parse as direct response (ignoring mode)
+        int r = CHECK_PARSER_URC(reader->scanf("+CREG: %*d,%d,\"%x\",\"%x\",%d", &val[0], &val[1], &val[2], &val[3]));
+        // Reparse as URC
+        if (r <= 1) { r = CHECK_PARSER_URC(reader->scanf("+CREG: %d,\"%x\",\"%x\",%d", &val[0], &val[1], &val[2], &val[3])); }
         CHECK_TRUE(r >= 1, SYSTEM_ERROR_UNKNOWN);
         // Home network or roaming
-        if (val[r - 1] == 1 || val[r - 1] == 5) {
+        if (val[0] == 1 || val[0] == 5) {
             self->creg_ = RegistrationState::Registered;
         } else {
             self->creg_ = RegistrationState::NotRegistered;
         }
         self->checkRegistrationState();
+        // Cellular Global Identity (partial)
```

Comment:
May want to add a if (r >= 4) check here, and/or ensure int val[4] is memset to 0 or give val[2] and val[3] some proper `unknown` defaults.

Reply:
Initialized `val` array...
```cpp
int val[4] = {-1,-1,-1,-1};
```

Reply:
👍